backend/habitat/viewsets.py

```diff
@@ -870,3 +1083,82 @@ def bathymetry_statistics(request):
             return Response([])
         else:
             return Response(bathymetry_stats)
+
+@action(detail=False)
+@api_view()
+@renderer_classes((JSONRenderer, ShapefileRenderer))
+def habitat_observations(request):
+    params = {k: v or None for k, v in request.query_params.items()}
+    boundary_type        = params.get('boundary-type')
+    network              = params.get('network')
+    park                 = params.get('park')
+    zone                 = params.get('zone')
+    zone_iucn            = params.get('zone-iucn')
+    provincial_bioregion = params.get('provincial-bioregion')
+    mesoscale_bioregion  = params.get('mesoscale-bioregion ')
+    realm                = params.get('realm')
+    province             = params.get('province')
+    ecoregion            = params.get('ecoregion')
+
+    global_archive = []
+    sediment = []
+    squidle = []
+
+    with connections['transects'].cursor() as cursor:
+        if boundary_type == 'amp':
+            cursor.execute(SQL_GET_AMP_HABITAT_OBS_GLOBALARCHIVE, [network, park, zone, zone_iucn])
+        elif boundary_type == 'imcra':
+            cursor.execute(SQL_GET_IMCRA_HABITAT_OBS_GLOBALARCHIVE, [provincial_bioregion, mesoscale_bioregion])
+        elif boundary_type == 'meow':
+            cursor.execute(SQL_GET_MEOW_HABITAT_OBS_GLOBALARCHIVE, [realm, province, ecoregion])
+        else:
+            raise Exception('Unhandled boundary type!')
+
+        try:
+            columns = [col[0] for col in cursor.description]
+            namedrow = namedtuple('Result', columns)
+            results = [namedrow(*row) for row in cursor.fetchall()]
+
+            global_archive = [row._asdict() for row in results]
+        except:
+            pass
```

Comment:
What's the thinking behind this?  Do we expect benign exceptions that we ignore, or is this just convenient?

(I'm not against it per-se, just wondering if any thought has gone into it, or it was just convenient at the time)

Reply:
Yeah, it's probably a good idea to not allow any benign exceptions. I wasn't really thinking. We'll just have it fail on any exception 👍